In [1]:
import pandas as pd
import numpy as np

# Display settings
pd.set_option("display.max_columns", None)
pd.set_option("display.max_colwidth", 150)

In [2]:
df=pd.read_csv(r"C:\Users\Login\OneDrive\Desktop\AniWise-Anime-Recommendation-System\data\raw\myanimelist_recommender_ready.csv")

In [3]:
df.head()

,mal_id,title,title_english,title_japanese,type,episodes,status,score,rank,popularity,members,favorites,synopsis,genres,themes,demographics,studios,year,url
0,1,Cowboy Bebop,Cowboy Bebop,カウボーイビバップ,TV,26.0,Finished Airing,8.75,46.0,42,1975882,86621,"Crime is timeless. By the year 2071, humanity has expanded across the galaxy, filling the surface of other planets with settlements like those on ...","Action, Award Winning, Sci-Fi","Adult Cast, Space",NaN,Sunrise,1998.0,https://myanimelist.net/anime/1/Cowboy_Bebop
1,5,Cowboy Bebop: Tengoku no Tobira,Cowboy Bebop: The Movie,カウボーイビバップ 天国の扉,Movie,1.0,Finished Airing,8.38,215.0,642,398122,1711,"Another day, another bounty—such is the life of the often unlucky crew of the Bebop. However, this routine is interrupted when Faye, who is chasin...","Action, Sci-Fi","Adult Cast, Space",NaN,Bones,NaN,https://myanimelist.net/anime/5/Cowboy_Bebop__Tengoku_no_Tobira
2,6,Trigun,Trigun,トライガン,TV,26.0,Finished Airing,8.22,378.0,260,803358,16863,"Vash the Stampede is the man with a $$60,000,000,000 bounty on his head. The reason: he's a merciless villain who lays waste to all those that opp...","Action, Adventure, Sci-Fi",Adult Cast,Shounen,Madhouse,1998.0,https://myanimelist.net/anime/6/Trigun
3,7,Witch Hunter Robin,Witch Hunter Robin,Witch Hunter ROBIN (ウイッチハンターロビン),TV,26.0,Finished Airing,7.23,3260.0,1950,123424,674,"Though hidden away from the general public, Witches—those with supernatural powers—have always existed in human societies. Neither numerous nor in...","Action, Drama, Mystery, Supernatural",Detective,NaN,Sunrise,2002.0,https://myanimelist.net/anime/7/Witch_Hunter_Robin
4,8,Bouken Ou Beet,Beet the Vandel Buster,冒険王ビィト,TV,52.0,Finished Airing,6.92,4819.0,5655,16308,16,"It is the dark century and the people are suffering under the rule of the devil, Vandel, who is able to manipulate monsters. The Vandel Busters ar...","Action, Adventure, Fantasy",NaN,Shounen,Toei Animation,2004.0,https://myanimelist.net/anime/8/Bouken_Ou_Beet


In [4]:
df.shape

(28707, 19)

In [5]:
df.columns

Index(['mal_id', 'title', 'title_english', 'title_japanese', 'type',
       'episodes', 'status', 'score', 'rank', 'popularity', 'members',
       'favorites', 'synopsis', 'genres', 'themes', 'demographics', 'studios',
       'year', 'url'],
      dtype='str')

In [6]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 28707 entries, 0 to 28706
Data columns (total 19 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   mal_id          28707 non-null  int64  
 1   title           28707 non-null  str    
 2   title_english   12343 non-null  str    
 3   title_japanese  28587 non-null  str    
 4   type            28634 non-null  str    
 5   episodes        27931 non-null  float64
 6   status          28707 non-null  str    
 7   score           18598 non-null  float64
 8   rank            21806 non-null  float64
 9   popularity      28707 non-null  int64  
 10  members         28707 non-null  int64  
 11  favorites       28707 non-null  int64  
 12  synopsis        23629 non-null  str    
 13  genres          22800 non-null  str    
 14  themes          16917 non-null  str    
 15  demographics    10725 non-null  str    
 16  studios         17083 non-null  str    
 17  year            6192 non-null   float64
 1

In [7]:
df.describe(include="all")

,mal_id,title,title_english,title_japanese,type,episodes,status,score,rank,popularity,members,favorites,synopsis,genres,themes,demographics,studios,year,url
count,28707.000000,28707,12343,28587,28634,27931.000000,28707,18598.000000,21806.000000,28707.000000,2.870700e+04,28707.000000,23629,22800,16917,10725,17083,6192.000000,28707
unique,NaN,28489,12034,27318,9,NaN,3,NaN,NaN,NaN,NaN,NaN,23160,929,1026,8,1871,NaN,28491
top,NaN,Wan Xiang Zhi Wang,Promise,おしりたんてい,TV,NaN,Finished Airing,NaN,NaN,NaN,NaN,NaN,No synopsis has been added for this series yet.\n\nClick here to update this information.,Comedy,Music,Kids,Toei Animation,NaN,https://myanimelist.net/anime/61563/Wan_Xiang_Zhi_Wang
freq,NaN,4,5,9,8358,NaN,27698,NaN,NaN,NaN,NaN,NaN,24,2695,3951,6809,880,NaN,4
mean,33687.396419,NaN,NaN,NaN,NaN,14.147148,NaN,6.397713,10865.145189,14248.875536,3.842745e+04,429.596684,NaN,NaN,NaN,NaN,NaN,2009.873224,NaN
std,19486.123233,NaN,NaN,NaN,NaN,47.300443,NaN,0.889745,6276.800645,8225.464407,1.658327e+05,4490.622485,NaN,NaN,NaN,NaN,NaN,13.188576,NaN
min,1.000000,NaN,NaN,NaN,NaN,1.000000,NaN,1.890000,1.000000,1.000000,0.000000e+00,0.000000,NaN,NaN,NaN,NaN,NaN,1961.000000,NaN
25%,14990.000000,NaN,NaN,NaN,NaN,1.000000,NaN,5.780000,5430.250000,7124.500000,2.350000e+02,0.000000,NaN,NaN,NaN,NaN,NaN,2004.000000,NaN
50%,37736.000000,NaN,NaN,NaN,NaN,2.000000,NaN,6.370000,10862.500000,14237.000000,1.085000e+03,1.000000,NaN,NaN,NaN,NaN,NaN,2013.000000,NaN
75%,50047.000000,NaN,NaN,NaN,NaN,13.000000,NaN,7.030000,16301.750000,21368.500000,9.154000e+03,17.000000,NaN,NaN,NaN,NaN,NaN,2019.000000,NaN


In [8]:
df.isnull().sum().sort_values(ascending=False)

year              22515
demographics      17982
title_english     16364
themes            11790
studios           11624
score             10109
rank               6901
genres             5907
synopsis           5078
episodes            776
title_japanese      120
type                 73
mal_id                0
title                 0
status                0
members               0
popularity            0
favorites             0
url                   0
dtype: int64

In [9]:
df.duplicated().sum()

np.int64(216)

## Conclusion

- Dataset contains 28,707 anime records with 19 features.
- Data types are mostly appropriate.
- Some metadata columns contain missing values, which is expected for real-world anime databases.
- The dataset includes 216 duplicate records that will be handled during preprocessing.
- Overall, the dataset is suitable for building a content-based anime recommendation system.